scripts/test/Muon/elemental_analysis_2/elemental_analysis_loadrun_presenter_single_run_test.py

```diff
@@ -0,0 +1,85 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+
+from unittest import mock
+from mantidqt.utils.qt.testing import start_qapplication
+from qtpy.QtWidgets import QApplication, QWidget
+
+from Muon.GUI.ElementalAnalysis2.load_widget.load_models import LoadRunWidgetModel
+from Muon.GUI.ElementalAnalysis2.load_widget.load_run_widget_presenter import LoadRunWidgetPresenterEA
+from Muon.GUI.Common.load_run_widget.load_run_view import LoadRunWidgetView
+from Muon.GUI.Common.test_helpers.context_setup import setup_context_for_ea_tests
+
+
+@start_qapplication
+class LoadRunWidgetPresenterTest(unittest.TestCase):
+    def wait_for_thread(self, thread_model):
+        if thread_model:
+            thread_model._thread.wait()
+            QApplication.sendPostedEvents()
+
+    def setUp(self):
+        # Store an empty widget to parent all the views, and ensure they are deleted correctly
+        self.obj = QWidget()
+
+        setup_context_for_ea_tests(self)
+
+        self.view = LoadRunWidgetView(parent=self.obj)
+        self.model = LoadRunWidgetModel(self.loaded_data, self.context)
+        self.presenter = LoadRunWidgetPresenterEA(self.view, self.model)
+
+    def tearDown(self):
+        self.obj = None
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # TESTS
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def test_model_and_view_initialized_to_contain_no_data(self):
+        self.assertEqual(self.presenter.run_list, [])
+        self.assertEqual(self.view.get_run_edit_text(), "")
+
+    def test_user_can_enter_a_run_and_load_it_in_single_run_mode(self):
```

Comment:
I think we should be mocking the view and model, then checking the function calls have the correct arguments.